include/RAJA/policy/openmp/forall.hpp

```diff
@@ -66,12 +67,15 @@ RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
                              Iterable&& iter,
                              Func&& loop_body)
 {
-#pragma omp parallel
-  {
-    using RAJA::internal::thread_privatize;
-    auto body = thread_privatize(loop_body);
-    forall_impl(InnerPolicy{}, std::forward<Iterable>(iter), body.get_priv());
-  }
+
+  RAJA::Region<RAJA::omp_parallel_region>([&](){
+
```

Comment:
@davidbeckingsale @trws , I hope this is what you meant. The lambda has to be pass by reference otherwise it breaks.

Reply:
I think that's what @trws meant. How does it break if you capture by value?

Reply:
I get compilation errors, in particular the first error is:  
error: no matching function for call to 'forward(const RAJA::nested::IterableTiler<const RAJA::TypedRangeStrideSegment<long int>&>&)'
       forall_impl(InnerPolicy{}, std::forward<Iterable>(iter), body.get_priv());

Reply:
You can capture just the `iter` by reference and it works. I'd defer to @trws on what the most efficient implementation is.